src/PluginsManager/Microsoft.Performance.Toolkit.Plugins.Runtime/Manager/PluginsSystem.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.Toolkit.Plugins.Core.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Core.Transport;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Common;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Discovery;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Extensibility;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Installation;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Manager
+{
+    /// <summary>
+    ///    Represents a plugins system that can be used to discover and install plugins.
+    /// </summary>
+    public sealed class PluginsSystem
+    {
+        /// <summary>
+        ///     Creates an instance of the <see cref="PluginsSystem"/>.
+        /// </summary>
+        /// <param name="installer">
+        ///     The installer to use.
+        /// </param>
+        /// <param name="discoverer">
+        ///     The discoverer to use.
+        /// </param>
+        private PluginsSystem(
```

Comment:
Wonder if we should leave this as public, but in docstring call out the static methods below

Reply:
Yep. Changed in the 2nd PR